.github/workflows/main.yml

```diff
@@ -0,0 +1,16 @@
+on: [push, pull_request]
+name: CI
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out source code
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+      - name: Install dependencies
+        run: |
+          pip install git+https://github.com/nmigen/nmigen.git#egg=nmigen
```

Comment:
We should test nmigen-boards against the last release of nMigen.

Reply:
Currently nmigen-boards requires a unreleased version of nMigen (atleast because of the `lattice_machxo2.py` -> `lattice_machxo_2_3l.py` rename / the addition of  machxo3l support).
Testing against the newest nMigen release would mean its currently broken till the nMign 0.3 release, is that intended?

Reply:
> Testing against the newest nMigen release would mean its currently broken till the nMign 0.3 release, is that intended?

Hmm, we don't have a procedure for this. I guess it's fine: it's no worse than what we had before (slightly better because you can look at CI results), and it will remind us to do something about this issue.